.github/workflows/build-test.yaml

```diff
@@ -0,0 +1,45 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 16 * * *' # Runs daily at 16:00 UTC
+
+permissions:
+  contents: read
+
+jobs:
+  build-test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        rmq-version: ['streams-tiered-storage']
+        include:
+          - rmq-version: 'streams-tiered-storage'
+            otp-version: 27
+            elixir-version: 1.18
+    steps:
+      - id: restore-rabbitmq-server-cache
+        uses: actions/cache/restore@v4
+        with:
+          path: ${{ github.workspace }}/rabbitmq-server
+          key: rmq-${{ matrix.rmq-version }}-otp-${{ matrix.otp-version }}-elixir-${{ matrix.elixir-version }}
+      - id: maybe-fail-rabbitmq-server-miss
+        if: steps.restore-rabbitmq-server-cache.outputs.cache-hit != 'true'
+        uses: actions/github-script@v8
+        with:
+          script: |
+            core.setFailed('could not restore cached rabbitmq-server build');
```

Comment:
Rather than failing the workflow I think we need to run the parts of the `build-rabbitmq-server.yaml` workflow that clone Rabbit. This workflow currently needs that workflow to run successfully in order to work

Reply:
I just blindly did what aws does there, wonder how it works on aws.

Reply:
actually yeah I am stupid I should just remove the test build in this PR, let this merge run the action to build the cache then add this.